src/main/java/duke/tool/UI.java

```diff
@@ -0,0 +1,75 @@
+package duke.tool;
+
+import java.util.Scanner;
+
+/** deals with interactions with the user */
+public class UI {
+  public static String UIString = "";
+  private static String indent = "    ";
+  private static String line = "    ____________________________________________________________";
+  private Storage storage;
+  private Scanner userScanner = new Scanner(System.in);
+
+  public UI(Storage storage) {
+    this.storage = storage;
+  }
+
+  /**
+   * Receives user command
+   *
+   * @param stringScanner Used to scan input string in testing
+   * @param isTest True in testing
+   * @return The command String of the user
+   */
+  public String readCommand(Scanner stringScanner, boolean isTest) {
+    if (stringScanner.hasNextLine()) {
+      return stringScanner.nextLine();
+    } else {
+      if (!isTest) {
+        return userScanner.nextLine();
+      } else {
+        return "bye";
+      }
+    }
+  }
+
+  public void printWelcomeMessage() {
+    String logo =
+        indent
```

Comment:
I think second indents should be 8 spaces (just like the following lines). Not 100% sure though, might want to double check.